test/unit/Server/Middleware/TransactionMarkerMiddlewareHelperTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * @author    Craig Gosman <craig@ingenerator.com>
+ * @licence   proprietary
+ */
+
+namespace unit\Server\Middleware;
+
+use Ingenerator\CloudTasksWrapper\Server\Middleware\TransactionMarkerMiddlewareHelper;
+use Ingenerator\PHPUtils\DateTime\DateTimeImmutableFactory;
+use PHPUnit\Framework\TestCase;
+use function uniqid;
+
+class TransactionMarkerMiddlewareHelperTest extends TestCase
+{
+    public function test_it_adds_header_options_to_task_array()
+    {
+        $task   = [
+            'anything' => 'something',
+            'options'  => ['headers' => ['X-CustomHeader' => 'foo']],
+        ];
+        $uuid   = uniqid();
+        $expiry = DateTimeImmutableFactory::atUnixSeconds(1570792394);
+        $this->assertSame(
+            [
+                'anything' => 'something',
+                'options'  => [
+                    'headers' => [
+                        'X-CustomHeader'       => 'foo',
+                        'X-Transaction'        => $uuid,
+                        'X-Transaction-Expire' => '2019-10-11 11:13:14',
+                    ],
+                ],
+            ],
+            TransactionMarkerMiddlewareHelper::addTransactionMarkerHeaders($task, $uuid, $expiry)
+        );
+    }
```

Comment:
Potentially worth adding a future-proofing test that the result is actually in the right place / format for the `CreateTaskOptions` to actually record it as being headers?

With that and the header-name consts you have (good call) the tests protect this against any changes to either the client-side or server-side related code.

Reply:
see d523430 ?